src/core/sharedlib.d

```diff
@@ -0,0 +1,43 @@
+module core.sharedlib;
```

Comment:
Shouldn't this reside in the `utils` package or being `private` until we are sure on the API exposed?

Reply:
done, after slack discussion, moved to core.internal.sharedlib ; and also moved other reusable utilities under core.internal